internal/middleware/middleware.go

```diff
@@ -40,6 +45,27 @@ func (s *wrappedServerStream) RecvMsg(m interface{}) error {
 		storeID := r.GetStoreId()
 		s.wrappedContext = context.WithValue(s.Context(), storeIDCtxKey, storeID)
 		fields = append(fields, zap.String(storeIDKey, storeID))
+
+		if r, ok := m.(hasGetAuthorizationModelID); ok {
+			modelID := r.GetAuthorizationModelId()
+
+			if modelID == "" {
+				if s.datastore == nil {
+					return errors.New("failed to retrieve model id")
+				}
```

Comment:
i'm not fond of this check... I think we can change the constructor to accept a datastore, and if `nil` is passed we use a NoOp Datastore?